File: gfg/POTD/2025-02-05/solution.js
/*
class Node{
    constructor(data){
        this.data = data;
        this.left = null;
        this.right = null;
    }
}
*/

class Solution {
	/**
	 * @param {Node} node
	 * @return {void}
	 */
	// Function to convert a binary tree into its mirror tree.
	mirror(node) {
		// your code here
		if (!node) return

		;[node.left, node.right] = [node.right, node.left]

		this.mirror(node.left)
		this.mirror(node.right)
	}
}

File: gfg/POTD/2025-02-05/solution.py
'''
class Node:
    def _init_(self, val):
        self.right = None
        self.data = val
        self.left = None
'''
# your task is to complete this function

class Solution:
    #Function to convert a binary tree into its mirror tree.
    def mirror(self, root):
        if not root:
            return
        
        # Swap left and right children
        root.left, root.right = root.right, root.left

        self.mirror(root.left)
        self.mirror(root.right)